airbyte/cloud/connections.py

```diff
@@ -258,6 +259,44 @@ def get_sync_result(
 
     # Deletions
 
+    def update_config(
+        self,
+        *,
+        name: str | None = None,
+        configurations: api_models.StreamConfigurationsInput | None = None,
+        schedule: api_models.AirbyteAPIConnectionSchedule | None = None,
+        prefix: str | None = None,
+        status: api_models.ConnectionStatusEnum | None = None,
+    ) -> CloudConnection:
```

Comment:
Let's decouple these operations...

1. Move rename operation into a dedicated "rename()" method. (This is not destructive.)
2. Consider removing the option to change the status - I don't think we have use cases for this. It seems doing so might "void the warrantee" so to speak.
3. Move schedule changes to a dedicated "update_schedule()" method. (This is not destructive.)

What's left in "update_config()" would be to accept configurations and optional prefix.